### PR TITLE
hotfix: Leave Allocation

### PR DESCRIPTION
**Pull request by mymi14s**
## Is this a Feature, Chore or Bug?
- [] Feature
- [x] Chore
- [] Bug

## Clearly and concisely describe the feature, chore or bug.
This PR assigns Leave Allocation to employees with non, it also ensure new employees are allocated leaves.

## Analysis and design (optional)
Analyse and attach the design documentation

## Solution description
Describe your code changes in detail for reviewers.

## Is there a business logic within a doctype?
    - [] Yes
    - [] No

## Output screenshots (optional)
Post the output screenshots, if a UI is affected or added due to this feature.

## Areas affected and ensured
List out the areas affected by your code changes.

## Is there any existing behavior change of other features due to this code change?
Mention Yes or No. If Yes, provide the appropriate explanation.

## Did you test with the following dataset?
- [] Existing Data
- [] New Data

## Was child table created?
    - [] is attachment required?
        did you test attachment
## Did you delete custom field?
    - [] Yes
    - [] No
        If yes, did you write a delete patch?

## Is patch required?
- [] Yes
- [] No
    ## Was the patch test?

## Which browser(s) did you use for testing?
  - [] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/hiring/utils.py [hunk 1]
```diff
@@ -314,24 +314,25 @@ def generate_employee_id(doc):
...
+        assignment.assignment_based_on = 'Joining Date'
+        assignment.leave_policy = doc.leave_policy
+        assignment.effective_from = doc.date_of_joining
+        # effective_to is an year of addition to effective_from
+        assignment.effective_to = getdate(add_days(add_years(doc.date_of_joining, 1), -1))
+        assignment.carry_forward = True
+        assignment.leaves_allocated = False
+        assignment.save()
+        assignment.submit()
+        grant_leave_alloc_for_employee(assignment)
```
Comment (pjamsheer) [inline one_fm/hiring/utils.py]: grant_leave_alloc_for_employee will be called on submit of leave policy assignment and it is customised on LeavePolicyAssignmentOverride

Comment (mymi14s) [reply one_fm/hiring/utils.py]: @pjamsheer that has not been allocating since last year.

Comment (mymi14s) [reply one_fm/hiring/utils.py]: Tested this and i worked, you can test at your end as well

Comment (pjamsheer) [reply one_fm/hiring/utils.py]: It is confirmed from the HR that, Leave Policy Assignment and Leave Allocation are creating on creation of Employee.
The method `create_leave_policy_assignment` is called from the after insert of employee record.

**Pull request merged**